lib/gingr/solr_indexer.rb

```diff
@@ -1,49 +1,45 @@
 # frozen_string_literal: true
 require 'faraday/net_http_persistent'
+require 'find'
 require 'rsolr'
 require_relative 'config'
 
 module Gingr
   class SolrIndexer
     include Config
 
-    attr_reader :reference_urls
-    attr_reader :solr
+    attr_accessor :reference_urls
+    attr_accessor :solr
 
-    def initialize(url, reference_urls = {})
-      @solr = RSolr.connect url:, adapter: :net_http_persistent
-      @reference_urls = reference_urls
+    def initialize(solr = nil, reference_urls = nil)
+      solr ||= ENV['SOLR_URL'] || Gingr::Config.default_options[:solr_url]
+      solr = RSolr.connect url: solr, adapter: :net_http_persistent if solr.kind_of? String
+      @solr = solr
```

Comment:
I prefer this sort of initialization pattern because it provides convenience (initializing via a connection string) while still supporting passing the Solr client directly (as is done in the tests).

Note that I've also moved the default value logic into this class where it's most relevant. That way, the CLI class can focus just on passing `options`, and no matter how this class is initialized it will always execute the SOLR_URL fallback logic.

Reply:
Sounds good. When no Environment variable is available,  ENV['SOLR_URL'] will through error,  probably need a fallback for ENV:  ENV['SOLR_URL', nill]

Reply:
What version of Ruby did you test that with? Running `irb` in our Docker environment (Ruby 3.2.2) it's fine:

```ruby
ENV['FOO'] || 'bar' # evaluates to 'bar' without error
```

Reply:
Oh, actually, I use ENV.fetch.  ENV.fetch['SOLR_URL'] will raise an error when SOLR_URL environment variable is not available.